specs/interop/predeploys.md

```diff
@@ -254,6 +279,62 @@ Note that the `relayMessage` function is `payable` to enable relayers to earn in
 To enable cross chain authorization patterns, both the `_sender` and the `_source` MUST be exposed via `public`
 getters.
 
+#### Sending Back Messages Hashes
+
+When sending back a message that failed to be relayed on the destination chain
+to the source chain, it's crucial to ensure the message can only be sent back
+to the `L2ToL2CrossDomainMessenger` contract in its source chain.
+
+```solidity
+function sendMessageHashBack(uint256 _messageSource, uint256 _nonce, address _sender, address _target, bytes calldata _message) external nonReentrant {
```

Comment:
So we need the preimages instead of just being able to pass the digest so that we can check that its not sending to self chain?

The invariant is held within

```solidity
function sendMessage(uint256 _destination, address _target, bytes calldata _message) external {
    require(_destination != block.chainid);
    ...
```

So i think we should be able to just pass the message hash and can save on calldata and on chain hashing

Reply:
I like the idea of calling this function `revert(bytes32)` or something more minimal

Another thought is that there is no auth on this so anybody can call it on behalf of somebody else, there is a grief here and the `RETURN_DELAY` is the only thing that gives time for replayability

Reply:
The preimage was required to know in which chain the initiating message originated, so the `source` argument was key here. The reason is that without it someone could send back the digest to the wrong chain. Hashing the preimage gives us a way to know the source corresponds to the resulting digest/`messageHash`. 

However, we agree that `revert(bytes32)` will be a lot cleaner, so we just pushed a change that makes it work by having a `FailedMessage` struct containing the `source` chain id of the failed message, as well as the timestamp in which it first failed to be relayed.

Yes, the reason behind the `RETURN_DELAY` is to give enough time to replay to the users so griefing is not possible immediately. We didn't a good way to add auth to this function without adding a parameter to the encoding of the `messageHash`, or without adding more complexity.

We also added more descriptions and comments in the spec, let us know if you would like them to be ordered/styled different.